src/index.ts

```diff
@@ -63,8 +68,11 @@ export async function findManyCursorConnection<Model = { id: string }, Cursor =
     if (hasPreviousPage) nodes.shift()
   } else {
     // Execute the underlying query operations
-    nodes = await findMany({})
-    totalCount = await aggregate()
+    nodes = await findMany({
+      take: topLevelFields === undefined || topLevelFields.includes('edges') ? undefined : 0,
```

Comment:
Add a small explanatory comment here:

`// If the edges were not in the request fields, do not load any nodes`